subiquity/server/server.py

```diff
@@ -533,6 +532,48 @@ async def start_site(self, runner: web.AppRunner):
     def base_relative(self, path):
         return os.path.join(self.base_model.root, path)
 
+    def load_cloud_config(self):
+        # cloud-init 23.3 introduced combined-cloud-config, which helps to
+        # prevent subiquity from having to go load cloudinit modules.
+        # This matters because a downgrade pickle deserialization issue may
+        # occur when the cloud-init outside the snap (which writes the pickle
+        # data) is newer than the one inside the snap (which reads the pickle
+        # data if we do stages.Init()).  LP: #2022102
+
+        # The stages.Init() code path should be retained until we can assume a
+        # minimum cloud-init version of 23.3 (when Subiquity drops support for
+        # Ubuntu 22.04.2 LTS and earlier, presumably)
+
+        cloud_cfg = get_host_combined_cloud_config()
+        if len(cloud_cfg) > 0:
```

Comment:
could just use truthy value of dicts to avoid calculating length of keys and then comparing to 0
```suggestion
        if cloud_cfg:
```

Reply:
Yes, but we tend to be more explicit on this point in Subiquity, so I'm matching codebase convention.